src/xunit.runner.uap/MainPage.xaml.cs

```diff
@@ -15,22 +14,40 @@ public sealed partial class MainPage : Page
         public MainPage()
         {
             this.InitializeComponent();
-            log = new StringBuilder();
+            Application.Current.UnhandledException += OnUnhandledException;
         }
 
-        internal static StringBuilder log;
-
-        public void UpdateLog(string text)
-        {
-            log.AppendLine(text);
-        }
+        internal static StreamWriter log;
 
         private void RunXunitTestsInDirectory(object sender, RoutedEventArgs e)
         {
             // Run tests for assemblies in current directory
             XunitTestRunner runner = new XunitTestRunner();
+            log = Helpers.GetStreamToFileInLocalStorage(@"AC\Temp\stdout.txt");
+
+            Task.Run(() => runner.RunTests(App.LaunchArgs.Arguments, log));
+        }
+
+        static void OnUnhandledException(object sender, UnhandledExceptionEventArgs e)
+        {
+            var ex = e.Exception as Exception;
```

Comment:
isn't it `e.ExceptionObject`

Reply:
For Windows runtime is `Exception` (https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.unhandledexceptioneventargs)

Reply:
My mistake. I think I'm using the wrong event handler. Will update :)